File: common/src/main/java/com/ultreon/mods/advanceddebug/init/ModDebugPages.java
package com.ultreon.mods.advanceddebug.init;

import com.ultreon.libs.commons.v0.Identifier;
import com.ultreon.mods.advanceddebug.AdvancedDebug;
import com.ultreon.mods.advanceddebug.api.client.menu.DebugPage;
import com.ultreon.mods.advanceddebug.client.menu.DebugGui;
import com.ultreon.mods.advanceddebug.client.menu.pages.*;

@SuppressWarnings("unused")
public class ModDebugPages {
    public static final DebugPage DEFAULT = register("default", new DefaultPage());
    public static final PlayerPage PLAYER = register("player", new PlayerPage());
    public static final BlockPage BLOCK = register("block", new BlockPage());
    public static final FluidPage FLUID = register("fluid", new FluidPage());
    public static final ItemPage ITEM = register("item", new ItemPage());
    public static final EntityPage ENTITY = register("entity", new EntityPage());
    public static final ItemEntityPage ITEM_ENTITY = register("item_entity", new ItemEntityPage());
    public static final LivingEntityPage LIVING_ENTITY = register("living_entity", new LivingEntityPage());
    public static final WorldPage WORLD = register("world", new WorldPage());
    public static final WorldInfoPage WORLD_INFO = register("world_info", new WorldInfoPage());
    public static final MinecraftPage MINECRAFT = register("minecraft", new MinecraftPage());
    public static final WindowPage WINDOW = register("window", new WindowPage());
    public static final ComputerPage COMPUTER = register("computer", new ComputerPage());

    private static <T extends DebugPage> T register(String name, T page) {
        return DebugGui.get().registerPage(new Identifier(AdvancedDebug.MOD_ID, name), page);
    }

    public static void init() {

    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/init/ModDebugPages.kt
package com.ultreon.mods.advanceddebug.init

import com.ultreon.mods.advanceddebug.AdvancedDebug
import com.ultreon.mods.advanceddebug.client.menu.DebugGui.Companion.get
import com.ultreon.mods.advanceddebug.client.menu.pages.*

@Suppress("unused")
object ModDebugPages {
    val player = get().registerPage(PlayerPage(AdvancedDebug.modId, "player"))
    val block = get().registerPage(BlockPage(AdvancedDebug.modId, "block"))
    val fluid = get().registerPage(FluidPage(AdvancedDebug.modId, "fluid"))
    val item = get().registerPage(ItemPage(AdvancedDebug.modId, "item"))
    val entity = get().registerPage(EntityPage(AdvancedDebug.modId, "entity"))
    val itemEntity = get().registerPage(ItemEntityPage(AdvancedDebug.modId, "item_entity"))
    val livingEntity = get().registerPage(LivingEntityPage(AdvancedDebug.modId, "living_entity"))
    val world = get().registerPage(WorldPage(AdvancedDebug.modId, "world"))
    val worldInfo = get().registerPage(WorldInfoPage(AdvancedDebug.modId, "world_info"))
    val minecraft = get().registerPage(MinecraftPage(AdvancedDebug.modId, "minecraft"))
    val window = get().registerPage(WindowPage(AdvancedDebug.modId, "window"))
    val computer = get().registerPage(ComputerPage(AdvancedDebug.modId, "computer"))
    fun init() {}
}